src/Helpers/Form/Template/Utils/Frontend.php

```diff
@@ -51,11 +51,10 @@ public static function getFormId() {
 			return $formId;
 		}
 
-		// Get form id from donor purchase session.
-		$donorSession = give_get_purchase_session();
-		$formId       = ! empty( $donorSession['post_data']['give-form-id'] ) ?
-			absint( $donorSession['post_data']['give-form-id'] ) :
-			null;
+		// Get form id on ajax request by donation id.
+		if ( ! empty( $_REQUEST['donation_id'] ) ) {
+			return absint( give_get_payment_form_id( absint( $_REQUEST['donation_id'] ) ) );
```

Comment:
Should check that `$_REQUEST['donation_id']` before attempting to get the payment form, so we can fail early.

Reply:
Not sure what you mean by this, @bradp, can you please provide a bit more detail?

Reply:
@bradp We think that any validation or escape must happen before using the function or class. They just made for doing things instead of doing validation or escaping.

cc: @JasonTheAdams

Reply:
If `$_REQUEST['donation_id']` is set to something like 'null' or 'a' or anything else, when it gets cast to an int, it probably wont be a valid form ID. Rather than attempt to get the form ID, we should first check that the absint is a value greater than 1.

Example:
```
if ( ! empty( $_REQUEST['donation_id'] ) && absint( $_REQUEST['donation_id'] ) {
    return absint( give_get_payment_form_id( absint( $_REQUEST['donation_id'] ) ) );
```

Reply:
@bradp I got it what you are saying. By mistake, I used two `absint`. Updating code  to

```
// Get form id on ajax request by donation id.
		if ( ! empty( $_REQUEST['donation_id'] ) ) {
			return give_get_payment_form_id( absint( $_REQUEST['donation_id'] ) );
		}
```

because `give_get_payment_form_id` will return `int`.

Reply:
@bradp fixed: https://github.com/impress-org/givewp/compare/5937c30f41b1...d205b6be44e0